packages/headless/src/modules/global-search/services/elastic.service.ts

```diff
@@ -0,0 +1,178 @@
+import { Product } from "src/entity/product";
+import { errorResponse, successResponse } from "src/utils/response";
+import { IServiceResponse } from "src/utils/response/service.response.interface"; 
+import { ISearchProductResponse } from "../rest/dto";
+import { productSearchSchema } from "./product.schema";
+
+@Injectable()
+export class ElasticService {
+  constructor(private readonly esService: ElasticsearchService, private readonly productSearchDB: ProductSrarchDatabase) {}
+
+
+  public async createIndex() {
+    try {
+      const index = searchConfig.index
+      await this.esService.indices.delete({ index });
+      await this.esService.indices.create({ index });
+    } catch (error) {
+      console.log(error)
+    }
+    
+  }
+
+  private async createSchema(){
+    await this.esService.indices.putMapping({ 
+      index: searchConfig.index, 
+      type: searchConfig.index,
+      include_type_name: true,
+      body: { 
+          properties: productSearchSchema
+      }
+  });
+  }
+  
+  private mapSearchData(e: Product): Record<string, any> { 
+      let data: any = {} 
+
+      data.infoProductId = e.id || ''
+      data.infoName = e?.info?.name || ''
+      data.infoShortDescription = e?.info?.shortDescription || ''
+      data.infoFullDescription = e?.info?.fullDescription || ''
+      data.infoSku = e?.info?.sku || ''
+      data.infoPrice = e?.info?.price || 0
+
+      data.metaKeywords = e?.meta?.keywords || []
+      data.metaTitle = e?.meta?.title || ''
+      data.metaDescription = e?.meta?.description || ''
+
+      data.brands = e?.brands || []
+      data.categories = e?.categories || []
+      data.manufacturer = e?.manufacturer  || {}
+      data.photos = e?.photos.map(p => p.url)  || []
+      data.tags = e?.tags || []
+
+      return data;
+  } 
+
+  async getSingleByElasticId(id: string): Promise<any>{
+    const {body} =  await this.esService.get({index: searchConfig.index, id})
+    return body._source
+  }
+
+  async singleInsert(productData: Product): Promise<number> { 
+    try { 
+      const index = searchConfig.index 
+      // if not exist
+      const body = this.mapSearchData(productData) 
+      const result = await this.esService.index({index,refresh: true, body}) 
+      return result.statusCode;
+    } catch (error) {
+      console.log(error)
+      return 0;
+    } 
+  }
+
+async deleteSingleByElasticId(id: string): Promise<number> { 
+    try { 
+      const index = searchConfig.index
+      const type = searchConfig.index 
+      const result = await this.esService.delete({ index, type,  id }) 
+      return result.statusCode;
+    } catch (error) {
+      console.log(error)
+      return 0;
+    } 
+  }
+
+  async bulkInsert(){
+      try {
+      await this.createIndex();
+      await this.createSchema();
+
+      const allProduct =  await this.productSearchDB.findAllProducts()
+      if(!allProduct || allProduct.length < 1){
+          return 0;
+      }
```

Comment:
don't insert all the data at a one go. rather paginate the product list and insert chunk by chunk. maybe you can insert 20 or 50 products per query

Reply:
bulkInsert() will be removed because we have already done this on db seeder.